replica-lib/src/accountsdb_repl_client.rs

```diff
@@ -0,0 +1,119 @@
+use {
+    log::*,
+    solana_sdk::clock::Slot,
+    std::{net::SocketAddr, sync::Arc},
+    tokio::runtime::Runtime,
+    tonic::{self, transport::Endpoint, Request},
+};
+
+tonic::include_proto!("accountsdb_repl");
+
+pub struct AccountsDbReplClient {
+    client: accounts_db_repl_client::AccountsDbReplClient<tonic::transport::Channel>,
+}
+
+#[derive(Debug)]
+pub enum ReplicaRpcError {
+    InvalidUrl(String),
+    ConnectionError(String),
+    GetSlotsError(String),
+    GetAccountsError(String),
+}
+
+impl From<tonic::transport::Error> for ReplicaRpcError {
+    fn from(err: tonic::transport::Error) -> Self {
+        ReplicaRpcError::ConnectionError(err.to_string())
+    }
+}
+
+impl AccountsDbReplClient {
+    pub async fn connect(rpc_peer: &SocketAddr) -> Result<Self, ReplicaRpcError> {
+        let url = format!("http://{}", rpc_peer);
```

Comment:
Do you know if we can use https here and get some added security (some TLS maybe)?

Reply:
I think the replication is mostly done between nodes trust each other in the same data center -- so communication between them is trusted. As for detecting data corruption in-transit -- I would look into the hashing of the account and see if we can piggyback on that. Using http would be introduce less CPU overhead and better performance.

Reply:
A TLS tunnel can also be added at the server ops level as well using existing off-the-shelf software.

Reply:
Noted. Thanks for the replies. Marking as resolved.